client/src/components/mapping/TableViewer/index.tsx

```diff
@@ -112,45 +115,59 @@ const TableViewer = ({ source }: Props) => {
           });
         });
     }
-  }, [resource, source.credential.owner, table, toaster]);
+  }, [resource, availableOwners, owner, table, toaster]);
 
   return (
     <div id="tableViewer">
+      <StringSelect
+        icon={'th'}
+        inputItem={owner?.name || ''}
+        items={availableOwners.map(o => o.name)}
+        maxItems={100}
+        onChange={(name: string) => {
+          setOwner(availableOwners.find(o => o.name === name)!);
+          setTable('');
+        }}
+      />
       <StringSelect
```

Comment:
We could disable this Select if owner is not selected

Reply:
done